rust/processor/src/processors/parquet_processors/parquet_objects_processor.rs

```diff
@@ -0,0 +1,243 @@
+            config.max_buffer_size,
+            config.parquet_upload_interval_in_secs(),
+        );
+
+        Self {
+            connection_pool,
+            config,
+            objects_sender,
+        }
+    }
+}
+
+impl Debug for ParquetObjectsProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ParquetObjectsProcessor {{ capacity of objects channel: {:?} }} ",
+            &self.objects_sender.capacity(),
+        )
+    }
+}
+
+#[async_trait]
+impl ProcessorTrait for ParquetObjectsProcessor {
+    fn name(&self) -> &'static str {
+        ProcessorName::ParquetObjectsProcessor.into()
+    }
+
+    async fn process_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+        start_version: u64,
+        end_version: u64,
+        _: Option<u64>,
+    ) -> anyhow::Result<ProcessingResult> {
+        let last_transaction_timestamp = transactions.last().unwrap().timestamp.clone();
+
+        let mut conn = self.get_conn().await;
+        let query_retries = self.config.query_retries;
+        let query_retry_delay_ms = self.config.query_retry_delay_ms;
+
+        // Moving object handling here because we need a single object
+        // map through transactions for lookups
+        let mut all_objects = vec![];
+        let mut all_current_objects = AHashMap::new();
+        let mut object_metadata_helper: ObjectAggregatedDataMapping = AHashMap::new();
+        let mut transaction_version_to_struct_count: AHashMap<i64, i64> = AHashMap::new();
+
+        for txn in &transactions {
+            let txn_version = txn.version as i64;
+            let timestamp = txn
+                .timestamp
+                .as_ref()
+                .expect("Transaction timestamp doesn't exist!");
+            #[allow(deprecated)]
+            let block_timestamp = chrono::NaiveDateTime::from_timestamp_opt(timestamp.seconds, 0)
+                .expect("Txn Timestamp is invalid!");
+            let changes = &txn
+                .info
+                .as_ref()
+                .unwrap_or_else(|| {
+                    panic!(
+                        "Transaction info doesn't exist! Transaction {}",
+                        txn_version
+                    )
+                })
+                .changes;
+
+            // First pass to get all the object cores
+            for wsc in changes.iter() {
+                if let Change::WriteResource(wr) = wsc.change.as_ref().unwrap() {
+                    let address = standardize_address(&wr.address.to_string());
+                    if let Some(object_with_metadata) =
+                        ObjectWithMetadata::from_write_resource(wr, txn_version).unwrap()
+                    {
+                        // Object core is the first struct that we need to get
+                        object_metadata_helper.insert(address.clone(), ObjectAggregatedData {
```

Comment:
I think you can do
ObjectAggregatedData {
   object: blabla,
   ...ObjectAggregatedData::default()
}